sideboard/websockets.py

```diff
@@ -635,6 +687,19 @@ def internal_action(self, action, client, callback):
         elif action is not None:
             log.warn('unknown action {!r}', action)
 
+    def clear_cached_response(self, client, callback):
+        """
+        As explained above, Sideboard caches the most recent response to a
+        subscription so that when we check the subscription we can see if new
+        data needs to be sent.  However, if the user makes a series of requests
+        with the same client/callback ids which return the same response, they
+        probably still expect to get data back.  This method is therefore called
+        every time we receive an explicit RPC call for a subscription to discard
+        the cached value, ensuring that an explicit RPC call to a service
+        exposed via websocket always receives a response.
+        """
+        self.cached_fingerprints[client].pop(callback, None)
```

Comment:
MAGFest probably doesn't care about this, so feel free to skip over this explanation, but just in case you want to know even more than what's described in the docstring about why we made this change, here's a fuller explanation:

Suppose a client sends the following websocket RPC call to the server:

```javascript
{
    "method": "ecards.get",
    "params": {
        "before": "2017-01-01"
    },
    "client": "client-1"
}
```

let's further suppose that what we get back is ``[]``, i.e. an empty list.

Then on the same websocket connection, we send the following:

```javascript
{
    "method": "ecards.get",
    "params": {
        "before": "2016-01-01"
    },
    "client": "client-1"
}
```

What would you intuitively expect to get back?  Presumably you'd expect to get back ``[]`` as well.  However, what Sideboard would actually return (before this change) in such a case is nothing.  Not an empty list, but you'd literally not get any RPC response!

Why is this?  Because Sideboard has response caching such that it doesn't send a subscription response if it's the same as the last time it sent a response on that subscription.  This is useful for efficiency, since it means that if one method has an ``@notifies`` which pushes out a check on all subscriptions, then we only actually push out responses if there's new data.

However, on further reflection, we probably want to invalidate this cache anytime the user explicitly makes an API call, since if you make a method call, you probably always want a response, even if the response isn't any different than the last time you made an API call on the same subscription.

That's what this change is all about.  MAGFest will start caring about this at the point where we start using Javascript to make WebSocket RPC requests.